### ISSUE TITLE
Review `Join_Condition.Between`

### ISSUE THREAD

**Issue by radeusgd** [enso-org/enso#7767]
As I [posted on Discord](https://discord.com/channels/401396655599124480/1148921098327564288/1149345804151160883) under a Bookclub:

> every time I try to use it in a bookclub (2-3rd time now), I wish it was flipped - the current behaviour is really un-intuitive to me
> 
> that is because, when I think of a join, I think of it in this way: left table is my 'main table' and I'm correlating 'additional data' from the right table (this 'left-orientation' mostly stems from the fact that left columns are kept as-is, whereas right columns, if clashing, get prefixed - so the left table is in a way privileged)
> 
> so e.g. here, I'd like to join to my left table some rows that fit in some range L <= x <= U
> but the Between operation does exactly the reverse! because the upper and lower limits are supposed to come from the right table, I need to treat the right one as the 'reference' for which I'll be adding values from left
> 
> to me this is always really hard to wrap my head around and makes using the join condition just quite more complicated, making me consider to just brute force it with cross_join and a filter
> 
> I think if we flipped the order (the bounds are in left table, and the value to match in-between comes from right) it would be much more intuitive. I'm not sure if tehre are use cases where the current form is better suited - if yes - maybe we should just have both?
> 
> I know technically it's equivalent - I can always reorder tables and change names - but it adds a significant cognitive load.

As @jdunkerley pointed out, there _are_ valid use cases for the current flow - in fact apparently many use cases flow naturally with this. But as I have shown, there are also use-cases where it is counter-intuitive.

Technically, the `join` is symmetric and table order can be flipped - but the left table is slightly privileged because its columns 'win' over the right one's in case of name clashes. So the symmetry is broken in terms of ergonomics.

-----

As part of this, we should look into if we can make a mirror version of `Join_Condition.Between`, where the `upper` and `lower` bounds come from the _left_ table and the value itself is checked in _right_ table.

Given that the current flow is also useful, we should not remove it. Instead we have to think how we can achieve both versions - maybe we should have two variants of the `Join_Condition.Between` constructor - one for each case?

We need to design what will work best.

**Comment by radeusgd** [comment]
This is put on hold until we get the new Join UI. Once that is done we can see how the problem persists and how we can propose a solution compatible with the new UI.

**Comment by jdunkerley** [comment]
I think design is correct so closing.
